restlib/restlib.go

```diff
@@ -193,3 +193,29 @@ func SetHeaders(w http.ResponseWriter, headerMap http.Header) {
 		}
 	}
 }
+
+// OnRestResultHTTPResult is called from generated code when an HTTP result is retrieved.
+func OnRestResultHTTPResult(ctx context.Context, result *HTTPResult, err error) {
+	if result == nil || result.HTTPResponse == nil {
```

Comment:
This logic isn't correct. If we only have an error then we want to propagate that also:
```go
var restResult *common.RestResult = nil
if result != nil && result.HTTPResponse != nil {
    restResult = ...
}
OnRestResult(ctx, restResult, err)
```